sqlmesh/core/snapshot/evaluator.py

```diff
@@ -652,7 +661,13 @@ def _create_snapshot(
                 )
                 try:
                     self.adapter.clone_table(target_table_name, snapshot.table_name(), replace=True)
-                    self.adapter.alter_table(target_table_name, tmp_table_name)
+                    alter_expressions = self.adapter.get_alter_expressions(
```

Comment:
Why do we check this here? This only happens in dev, why should we fail in dev?

Reply:
Oh yeah, i forgot that cloning only happens in dev - is the `snapshot.previous_versions` clause how this conditions on dev env?

```
            if (
                snapshot.is_forward_only
                and snapshot.is_materialized
                and snapshot.previous_versions
                and self.adapter.SUPPORTS_CLONING
            ):
```